pkg/check/datastore.go

```diff
@@ -226,16 +234,111 @@ func getPolicy(ctx *CheckContext, name string) ([]types.BasePbmProfile, error) {
 	klog.V(4).Infof("Checking data store %q with potential volume Name %s", dsName, volumeName)
 	if err := checkVolumeName(volumeName); err != nil {
 		return fmt.Errorf("datastore %s: %s", dsName, err)
 	}
+	if err := checkForDatastoreCluster(ctx, dsName); err != nil {
+		return err
+	}
 	return nil
 }
 
+func checkForDatastoreCluster(ctx *CheckContext, dataStoreName string) error {
+	tctx, cancel := context.WithTimeout(ctx.Context, *Timeout)
+	defer cancel()
+	finder := find.NewFinder(ctx.VMClient, false)
+	datacenters, err := finder.DatacenterList(tctx, "*")
+	if err != nil {
+		klog.Errorf("error listing datacenters: %v", err)
+		return nil
+	}
+	workspaceDC := ctx.VMConfig.Workspace.Datacenter
+	var matchingDC *object.Datacenter
+	for _, dc := range datacenters {
+		if dc.Name() == workspaceDC {
+			matchingDC = dc
+		}
+	}
+
+	// lets fetch the datastore
+	finder = find.NewFinder(ctx.VMClient, false)
+	finder.SetDatacenter(matchingDC)
+	tctx, cancel = context.WithTimeout(ctx.Context, *Timeout)
+	defer cancel()
+	ds, err := finder.Datastore(tctx, dataStoreName)
+	if err != nil {
+		klog.Errorf("error getting datastore %s: %v", dataStoreName, err)
+		return nil
+	}
+
+	var dsMo mo.Datastore
+	pc := property.DefaultCollector(matchingDC.Client())
+	properties := []string{DatastoreInfoProperty, SummaryProperty}
+	tctx, cancel = context.WithTimeout(ctx.Context, *Timeout)
+	defer cancel()
+	err = pc.RetrieveOne(tctx, ds.Reference(), properties, &dsMo)
+
+	if err != nil {
+		klog.Errorf("error getting properties of datastore %s: %v", dataStoreName, err)
+		return nil
+	}
+
+	// list datastore cluster
+	m := view.NewManager(ctx.VMClient)
+	kind := []string{"StoragePod"}
+	tctx, cancel = context.WithTimeout(ctx.Context, *Timeout)
+	defer cancel()
+	v, err := m.CreateContainerView(tctx, ctx.VMClient.ServiceContent.RootFolder, kind, true)
+	if err != nil {
+		klog.Errorf("error listing datastore cluster: %+v", err)
+		return nil
+	}
+	var content []mo.StoragePod
+	tctx, cancel = context.WithTimeout(ctx.Context, *Timeout)
+	defer cancel()
+	err = v.Retrieve(tctx, kind, []string{SummaryProperty, "childEntity"}, &content)
+	if err != nil {
+		klog.Errorf("error retrieving datastore cluster properties: %+v", err)
+		return nil
+	}
+	err = v.Destroy(tctx)
+	if err != nil {
+		klog.Errorf("error destroying view: %+v", err)
+		return nil
+	}
+	for _, ds := range content {
+		for _, child := range ds.Folder.ChildEntity {
+			tDS, err := getDatastore(ctx, child)
+			if err != nil {
+				klog.Errorf("fetching datastore %s failed: %v", child.String(), err)
+				continue
+			}
+			if tDS.Summary.Url == dsMo.Summary.Url {
+				return fmt.Errorf("datastore %s is part of %s datastore cluster", tDS.Summary.Name, ds.Summary.Name)
+			}
+		}
+	}
+	klog.V(2).Infof("Checked datastore %s for SRDS - no problems found", dataStoreName)
```

Comment:
This is called for every StorageClass  and might be too noisy.

Reply:
increased the log level.

Reply:
You increased it at wrong place, see above

Reply:
yikes. fixed.